pymwp/matrix.py

```diff
@@ -172,3 +189,80 @@ def resize(matrix: List[List[Polynomial]], new_size: int) -> List[List[Polynomia
         for j in range(bound):
             res[i][j] = matrix[i][j]
     return res
+
+
+def show(matrix: List[List[Any]], **kwargs) -> None:
+    """Pretty print a matrix at the screen.
+
+    Using the keyword arguments it is possible display additional text
+    before or after the matrix.
+
+    Args:
+        matrix: the matrix to display.
+        **prefix (str): display some text before displaying matrix
+        **postfix (str): display some text after displaying matrix
+    """
+    if 'prefix' in kwargs:
+        print(kwargs['prefix'])
+    for row in matrix:
+        print([str(r) for r in row])
+    if 'postfix' in kwargs:
+        print(kwargs['postfix'])
+    print(' ')
+
+
+def equals(matrix1: List[List[Any]], matrix2: List[List[Any]]) -> bool:
+    """Determine if two matrices are equal.
+
+    This function performs element-wise equality comparisons on values of
+    two matrices. The two matrices must be the same size. For any two matrices
+    of different size the result is always `False`.
+
+    This function can evaluate values that are comparable by equals `==`
+    operator.
+
+    Arguments:
+        matrix1: first matrix.
+        matrix2: second matrix.
+
+    Raises:
+        TypeError: If the matrix value is not iterable
+
+    Returns:
+        `True` if matrices are equal element-wise and `False` otherwise.
+    """
+    # equal size
+    if [len(row) for row in matrix1] != [len(row) for row in matrix2]:
+        return False
+
+    # element-wise comparison
+    for row_index, column in enumerate(matrix1):
+        for col_index, value in enumerate(column):
+            if matrix2[row_index][col_index] != value:
+                return False
+
+    return True
+
+
+def fixpoint(matrix: List[List[Any]]) -> List[List[Any]]:
```

Comment:
So I guess now the question may be "Do we need both this method _and_  [def fixpoint(self) -> Relation](https://github.com/seiller/pymwp/blob/4fe2d96d48d879f1d1e085583dab1dd0447d2e65/pymwp/relation.py#L208) ?

Reply:
Fixpoint in relation does homogenization also. This function assumes you have two matrices of same size. I would say, they have different use cases.